src/test/test-path-util.c

```diff
@@ -221,18 +227,32 @@ static void test_find_executable_full(void) {
 
         assert_se(unsetenv("PATH") == 0);
 
-        assert_se(find_executable_full("sh", NULL, true, &p, NULL) == 0);
+        assert_se(find_executable_full("sh", NULL, NULL, true, &p, NULL) == 0);
         puts(p);
         assert_se(streq(basename(p), "sh"));
         free(p);
 
-        assert_se(find_executable_full("sh", NULL, false, &p, NULL) == 0);
+        assert_se(find_executable_full("sh", NULL, NULL, false, &p, NULL) == 0);
         puts(p);
         assert_se(streq(basename(p), "sh"));
         free(p);
 
         if (oldpath)
                 assert_se(setenv("PATH", oldpath, true) >= 0);
+
+        _cleanup_(unlink_tempfilep) char fn[] = "/tmp/test-XXXXXX";
+        assert_se(fd = mkostemp_safe(fn));
+        (void) fchmod(fd, 0755);
+
+        test_file_name = strchr(fn, delim);
+
+        assert_se(find_executable_full(test_file_name, NULL, exec_search_paths, false, &p, NULL) == 0);
+        puts(p);
+        assert_se(streq(p, fn));
+        free(p);
+
+        unlink(fn);
```

Comment:
I don't think you need this since you declare it with  `_cleanup_(unlink_tempfilep)`

Reply:
This is necessary. Once fn is unlinked the following assert_se call is a negative test on a file that no longer exists.

Reply:
well, then you don't need the `_cleanup_` stuff at all. 

bu maybe wite this:

```c
fn = unlink_and_free(fn);
```

Reply:
I went with unlink without the free. I believe the free attempts to free an object that no longer exists because the unlink deletes the file.